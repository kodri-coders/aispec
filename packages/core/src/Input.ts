import { type XMLBaseConstructor } from './XMLBase';

interface IInputConstructor extends XMLBaseConstructor {
  name: string;
  description: string;
  schema: string;
}

export class Input {
  name: string;
  description: string;
  schema: string;

  constructor(public input: IInputConstructor) {
    this.name = input.name;
    this.description = input.description;
    this.schema = JSON.parse(input.schema);
  }
};
