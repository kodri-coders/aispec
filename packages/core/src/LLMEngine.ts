import { generateText, tool } from 'ai';
interface Model {
  name: string;
  temperature: number;
  max_tokens: number;
}
export class LLMEngine {
  model: Model;
  constructor(model: Model){
    this.model = model;
  }
  modelFactory(model: Model): LLMEngine {
    return new LLMEngine(model);
  }
  prepareRequest({
    systemPrompt,
    prompt,
    tools,
    model,
  }: {
    systemPrompt: string;
    prompt: string;
    tools: any;
    model: any;
  }): any {
    switch (model.name) {
      case 'gpt-4':
      case 'gpt-4o':
      case 'gpt-3.5':
        return {
          system: systemPrompt,
          prompt,
          tools,
          model: model.name,
          temperature: model.temperature,
          maxTokens: model.max_tokens,
        };
      case 'o1-mini':
      case 'o1-preview':
        return {
          messages: [
            {
              role: 'user',
              content: systemPrompt
            },
            {
              role: 'user',
              content: prompt
            }
          ],
          tools,
          model: model.name,
        };
      default:
        throw new Error('Model not supported');
    }
  }
  async generateText({
    prompt,
    systemPrompt,
  }: {
    prompt: string;
    systemPrompt: string;
  }): Promise<string> {
    const response = await generateText(
      this.prepareRequest({
        systemPrompt,
        prompt,
        tools: [],
        model: this.model,
      }));
    return response.text;
  }
}