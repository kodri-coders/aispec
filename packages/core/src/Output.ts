import { type XMLBaseConstructor } from './XMLBase';

interface IOutputConstructor extends XMLBaseConstructor {
  description: string;
  name: string;
  schema: string;
}

export class Output {
  description: string;
  name: string;
  schema: string;
  node: IOutputConstructor;

  constructor(public output: IOutputConstructor) {
    this.name = output.name;
    this.description = output.description;
    this.node = output;
    try {
      this.schema = JSON.parse(output.schema);
    } catch (error) {
      this.schema = output.schema;
    }
  }
};
