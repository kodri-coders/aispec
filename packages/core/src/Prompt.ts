export class Prompt {
  public variables: string[];
  private prompt: string;

  constructor(prompt: string) {
    this.prompt = prompt;
    this.variables = this.extractVariables(prompt);
  }

  extractVariables(prompt?: string): string[] {
    const variables = (prompt || this.prompt).match(/\$\{(.+?)\}/g);
    return (variables || []).map(variable => variable.replace(/\${|}/g, ''));
  }

  interpolate(data: any): string {
    return this.prompt.replace(/\${(.+?)}/g, (_, key) => {
      let value = '()';
      key.split('.').forEach((keyPart: string) => {
        value = data[keyPart];
      });
      return value;
    });
  }
};
