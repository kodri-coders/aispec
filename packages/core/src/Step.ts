import { Input } from './Input';
import { Prompt } from './Prompt';
import { XMLBase, type XMLBaseConstructor } from './XMLBase';
import { Assistant } from './Assistant';
import { Output } from './Output';

interface IStepConstructor extends XMLBaseConstructor {
  name: string;
  description: string;
  input: any;
  prompt: string;
  outputs: any;
  $ref?: string;
}

interface StepInterface {
  name: string;
  description: string;
  output: Output;
  prompt: Prompt;
  excludedTags: string[];
} 
export class Step extends XMLBase {
  name?: string;
  description?: string = '';
  output?: Output;
  prompt?: Prompt;
  excludedTags: string[] = ['model'];
  mapping: any= {
    id: "@id",
    name: 'name',
    description: 'description',
    input: ['input', Input],
    prompt: ['prompt', Prompt],
    output: ['output', Output],
    model: 'model'
  }
  constructor(
    public step: IStepConstructor,
  ) {
    super(step);
    this.mount();
  }

  getPrompt(input?: any): any {
    return this.prompt?.interpolate(input);
  }
};
