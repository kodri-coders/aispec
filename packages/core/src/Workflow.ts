import { Assistant } from './Assistant';
import { Step } from './Step';
import { XMLBase, type XMLBaseConstructor } from './XMLBase';

interface IWorkflowConstructor extends XMLBaseConstructor {
  '@id': string;
  name: string;
  description: string;
  steps: {
    step: {
      name: string;
      description: string;
      input: any;
      prompt: string;
      outputs: any;
    }[];
  };
}

export class Workflow extends XMLBase {
  name: string='';
  description: string='';
  steps: Step[] = [];
  excludedTags: string[] = ['model'];
  mapping: any= {
    id: "@id",
    name: 'name',
    description: 'description',
    steps: ['steps.step', Step],
    model: 'model'
  }
  constructor(
    public workflow: IWorkflowConstructor,
  ) {
    super(workflow);
    this.mount();
  }

  run(input?: any): any {
    return this.steps.map((step: any) => step.runStep(input));
  }
};
