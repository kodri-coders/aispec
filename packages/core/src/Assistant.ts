import { Skill } from './Skill';
import { Workflow } from './Workflow';
import { WorkflowRunner } from './WorkflowRunner';
import { XMLBase, type XMLBaseConstructor } from './XMLBase';

interface IAssistantConstructor extends XMLBaseConstructor {
  '@id'?: string;
  name?: string;
  description?: string;
  skills?: {
    skill: any[];
  };
  workflows?: {
    workflow: any[];
  };
}

export class Assistant extends XMLBase {
  name?: string;
  description?: string;
  workflows?: Workflow[] = [];
  skills?: Skill[] = [];

  constructor(assistant: IAssistantConstructor | XMLBase) {
    super(assistant);
    this.name = this.node.name;
    this.description = this.node.description;
    this.model = this.node.model;

    this.skills = this.node.skills?.skill.map((skill: any) => {
      return new Skill(skill, this);
    });

    this.workflows = this.node.workflows?.workflow.map((workflow: any) => {
      return new Workflow(workflow);
    });
  }

  findWorkflow(workflowId: string): any {
    let targetWorkflow: Workflow | undefined;
    let containingSkill: Skill | undefined;

    // Check global workflows
    targetWorkflow = this.workflows?.find(
      (w) => w.workflow['@id'] === workflowId,
    );

    // If not found in global workflows, check skills
    if (!targetWorkflow) {
      this.skills?.forEach((skill) => {
        const found = skill.workflows?.find(
          (w) => w.workflow['@id'] === workflowId,
        );
        if (found) {
          targetWorkflow = found;
          containingSkill = skill;
        }
      });
    }

    if (!targetWorkflow) {
      throw new Error(`Workflow with id ${workflowId} not found`);
    }
    return { workflow: targetWorkflow, skill: containingSkill };
  }

  loadWorkflow(workflowId: string, input?: any): any {
    // Find the workflow and its containing skill
    const { workflow: targetWorkflow, skill: containingSkill } =
      this.findWorkflow(workflowId);
    const xmlObj: any = {
      assistant: {
        name: this.name,
        description: this.description,
        skills: {
          skill: [],
        },
      },
    };

    // Add global skills
    if (this.skills) {
      xmlObj.assistant.skills.skill = this.skills
        .filter((s) => s !== containingSkill)
        .map((s) => ({ name: s.name, description: s.description }));
    }
    if (targetWorkflow) {
      xmlObj.assistant.workflows = {
        workflow: [targetWorkflow],
      };
    }
    // Add the skill containing the workflow
    if (containingSkill) {
      xmlObj.assistant.skills.skill.push(containingSkill.node);
    }

    return new WorkflowRunner(new Assistant(xmlObj), targetWorkflow);
  }
}
