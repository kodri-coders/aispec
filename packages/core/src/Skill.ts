import { Assistant } from './Assistant';
import { Workflow } from './Workflow';
import { XMLBase, type XMLBaseConstructor } from './XMLBase';

interface ISkillConstructor extends XMLBaseConstructor {
  '@id': string;
  name: string;
  description: string;
  workflows?: {
    workflow: any[];
  };
}

export class Skill extends XMLBase {
  workflows?: Workflow[] = [];
  excludedTags: string[] = ['model'];
  skills?: Skill[] = [];
  mapping: any= {
    id: "@id",
    workflows: ['workflows.workflow', Workflow],
    skills: ['skills.skill', Skill],
    description: 'description',
    name: 'name'
  }
  constructor(
    public skill: ISkillConstructor,
  ) {
    super(skill);
    this.mount();
  }
};
