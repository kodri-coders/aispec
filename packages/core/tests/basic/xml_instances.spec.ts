import * as path from 'path';

import { Assistant } from '../../src';

const artifactPath = path.join(
  __dirname,
  'artifacts',
  'movie-builder-assistant.xml',
);
const assistant = new Assistant({ '@ref': artifactPath });
describe('assistant', () => {
  it('loads name and description', () => {
    expect(assistant.name).toMatchSnapshot();
    expect(assistant.description).toMatchSnapshot();
  });
  it('loads workflows', () => {
    expect(assistant.workflows?.length).toMatchSnapshot();
  });
  it('starts workflow', async () => {
    const workflowRunner = assistant.loadWorkflow('character-building');
    const mockResponses: any = {
      'Produce 2 surnames that sound good with the name: John': {
        surnames: ['Doe', 'Smith', 'Lan'],
      },
      'Given the name: John select one of the surnames: Doe,Smith,Lan': {
        surname: 'Doe',
      },
      'Generate 10 backstories for the character: John Doe': {
        backstories: [
          'Backstory 1',
          'Backstory 2',
          'Backstory 3',
          'Backstory 4',
          'Backstory 5',
          'Backstory 6',
          'Backstory 7',
          'Backstory 8',
          'Backstory 9',
          'Backstory 10',
        ],
      },
      'Enhance the backstory: Backstory 1': {
        title: 'Backstory 1',
        story: 'Enhanced backstory 1',
      },
      'Enhance the backstory: Backstory 2': {
        title: 'Backstory 2',
        story: 'Enhanced backstory 2',
      },
      'Enhance the backstory: Backstory 3': {
        title: 'Backstory 3',
        story: 'Enhanced backstory 3',
      },
      'Enhance the backstory: Backstory 4': {
        title: 'Backstory 4',
        story: 'Enhanced backstory 4',
      },
      'Enhance the backstory: Backstory 5': {
        title: 'Backstory 5',
        story: 'Enhanced backstory 5',
      },
      'Enhance the backstory: Backstory 6': {
        title: 'Backstory 6',
        story: 'Enhanced backstory 6',
      },
      'Enhance the backstory: Backstory 7': {
        title: 'Backstory 7',
        story: 'Enhanced backstory 7',
      },
      'Enhance the backstory: Backstory 8': {
        title: 'Backstory 8',
        story: 'Enhanced backstory 8',
      },
      'Enhance the backstory: Backstory 9': {
        title: 'Backstory 9',
        story: 'Enhanced backstory 9',
      },
      'Enhance the backstory: Backstory 10': {
        title: 'Backstory 10',
        story: 'Enhanced backstory 10',
      },
      "Generate 10 backstories for the character: John Doe": {
        backstories: [
          "Backstory 1",
          "Backstory 2",
          "Backstory 3",
          "Backstory 4",
          "Backstory 5",
          "Backstory 6",
          "Backstory 7",
          "Backstory 8",
          "Backstory 9",
          "Backstory 10",
        ],
      },
      "Enhance the backstory: Backstory 1": {
        title: "Backstory 1",
        story: "Enhanced backstory 1",
      },
      "Enhance the backstory: Backstory 2": {
        title: "Backstory 2",
        story: "Enhanced backstory 2",
      },
      "Enhance the backstory: Backstory 3": {
        title: "Backstory 3",
        story: "Enhanced backstory 3",
      },
      "Enhance the backstory: Backstory 4": {
        title: "Backstory 4",
        story: "Enhanced backstory 4",
      },
      "Enhance the backstory: Backstory 5": {
        title: "Backstory 5",
        story: "Enhanced backstory 5",
      },
      "Enhance the backstory: Backstory 6": {
        title: "Backstory 6",
        story: "Enhanced backstory 6",
      },
      "Enhance the backstory: Backstory 7": {
        title: "Backstory 7",
        story: "Enhanced backstory 7",
      },
      "Enhance the backstory: Backstory 8": {
        title: "Backstory 8",
        story: "Enhanced backstory 8",
      },
      "Enhance the backstory: Backstory 9": {
        title: "Backstory 9",
        story: "Enhanced backstory 9",
      },
      "Enhance the backstory: Backstory 10": {
        title: "Backstory 10",
        story: "Enhanced backstory 10",
      },
    };
    workflowRunner.callLLM = jest
      .fn()
      .mockImplementation((systemPrompt: string, prompt: string) => {
        console.log('prompt', prompt);
        return mockResponses[prompt];
      });
    workflowRunner.on('step-finished', (prompt: string, data: any) => {
      console.log('step finished', prompt, data);
      workflowRunner.next();
    });
    workflowRunner.on('workflow-finished', (workflow: any) => {
      console.log('workflow finished');
    });
    const workflow = workflowRunner.start({
      name: 'John',
      surnamesLength: 2,
    });
    expect(workflowRunner.assistant.toString()).toMatchSnapshot();

    await workflowRunner.onFinish();
    expect(workflowRunner.context).toMatchSnapshot();

    expect(workflowRunner.callLLM).toHaveBeenCalledTimes(14);
    expect(workflowRunner.logs).toMatchSnapshot();
  });
});
