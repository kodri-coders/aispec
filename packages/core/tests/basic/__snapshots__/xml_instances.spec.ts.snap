// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assistant loads name and description 1`] = `"Movie builder"`;

exports[`assistant loads name and description 2`] = `"Assistant 1 description"`;

exports[`assistant loads workflows 1`] = `1`;

exports[`assistant starts workflow 1`] = `
"<name>Movie builder</name>
<description>Assistant 1 description</description>
<skills>
  <skill id="movie-building-skill">
    <name>Movie building skill</name>
    <description>Building a movie scenario means
            creating a story that takes place in
            a world with certain characters, 
            plot lines and themes. 
            The world can be a fantasy world, 
            a historical period, a real world city, 
            or any other setting imaginable. 
            The story can be a romance, a thriller, 
            a sci-fi epic or any other genre. The characters can be humans, animals, aliens, or any other type of being. The plot can be a simple one, like a romance that takes place in a small town, or a complex one, like a multi-layered epic that takes place in a fictional world. The themes can be simple, like love and friendship, or complex, like the nature of reality and the human condition. The movie scenario building skill allows you to explore all of these possibilities and create a unique story that expresses your imagination and creativity.</description>
  </skill>
  <skill id="world-building-skill">
    <name>World building</name>
    <description>A skill that allows you to build a world</description>
    <skills>
      <skill id="COT">
        <name>Chain of thought</name>
        <description>Understand the Question: 
    - Carefully read the input to grasp its full meaning and intent.
    - Identify Key Components: Break down the question into main parts or sub-questions that need to be addressed.
    - Gather Relevant Information: Recall relevant knowledge, facts, or data that pertain to each component.
    - Analyze Step-by-Step: Methodically examine each part, explaining the reasoning and logic behind each step.
    - Synthesize the Response: Combine the analyzed parts into a coherent and comprehensive answer.
    - Review for Clarity: Ensure the final response is clear, concise, and logically structured.

    When responding, first thoroughly understand the question. 
    Break it down into key components, gather relevant information 
    for each part, analyze each step logically, and then synthesize 
    your findings into a clear, comprehensive answer. 
    Explain your reasoning at each stage to ensure clarity and transparency 
    in your response.
    
    !IMPORTANT ALWAYS wrap your thought process in a \`reasoning\` xml tag</description>
      </skill>
    </skills>
    <workflows>
      <workflow id="character-building">
        <name>Character building</name>
        <description>A workflow that allows you to build a character with a given name</description>
        <input>
          <name>Human name</name>
          <description>Input description</description>
          <schema>{
            &quot;type&quot;: &quot;object&quot;,
            &quot;properties&quot;: {
              &quot;name&quot;: {
                &quot;type&quot;: &quot;string&quot;
              },
              &quot;surnamesLength&quot;: {
                &quot;type&quot;: &quot;number&quot;
              }
            }
          }</schema>
        </input>
        <steps>
          <step id="surname-generation">
            <model>
              <name>gpt-4o</name>
              <temperature>0.5</temperature>
              <max_tokens>1000</max_tokens>
            </model>
            <name>Surname generation</name>
            <description>Surname generation step</description>
            <prompt>Produce \${surnamesLength} surnames that sound good with the name: \${name}</prompt>
            <output id="surnames">
              <name>Surnames</name>
              <schema>{
                &quot;type&quot;: &quot;object&quot;,
                &quot;properties&quot;: {
                  &quot;surnames&quot;: {
                    &quot;type&quot;: &quot;array&quot;,
                    &quot;items&quot;: {
                      &quot;type&quot;: &quot;string&quot;
                    }
                  }
                }
              }</schema>
            </output>
          </step>
          <step id="surname-selection">
            <name>Surname selection</name>
            <description>Surname selection step</description>
            <prompt>Given the name: \${name} select one of the surnames: \${surnames}</prompt>
            <output id="surname">
              <name>Surname</name>
              <schema>{
                &quot;type&quot;: &quot;object&quot;,
                &quot;properties&quot;: {
                  &quot;surname&quot;: {
                    &quot;type&quot;: &quot;string&quot;
                  }
                }
              }</schema>
            </output>
          </step>
          <step id="character-generation">
            <name>Character generation</name>
            <description>Character generation step</description>
            <prompt>Generate a character with the name: \${name} and the surname: \${surname}</prompt>
            <output id="character">
              <name>Character</name>
              <schema>{
                &quot;type&quot;: &quot;object&quot;,
                &quot;properties&quot;: {
                  &quot;character&quot;: {
                    &quot;type&quot;: &quot;string&quot;
                  }
                }
              }</schema>
            </output>
          </step>
          <step id="backstories-generation">
            <name>Backstories generation</name>
            <description>Backstories generation step</description>
            <prompt>Generate 10 backstories for the character: \${character}</prompt>
            <output id="backstories">
              <name>Backstories</name>
              <schema>{
                &quot;type&quot;: &quot;object&quot;,
                &quot;properties&quot;: {
                  &quot;backstories&quot;: {
                    &quot;type&quot;: &quot;array&quot;,
                    &quot;items&quot;: {
                      &quot;type&quot;: &quot;string&quot;
                    }
                  }
                }
              }</schema>
            </output>
          </step>
          <step id="backstory-enhancement" loop="backstories" as="backstory">
            <name>Backstory enhancement</name>
            <description>Backstory enhancement step</description>
            <prompt>Enhance the backstory: \${backstory}</prompt>
            <output id="enhanced-backstories">
              <name>Enhanced backstory</name>
              <schema push="enhanced_backstories">{
                &quot;type&quot;: &quot;object&quot;,
                &quot;properties&quot;: {
                  &quot;title&quot;: {
                    &quot;type&quot;: &quot;string&quot;
                  },
                  &quot;story&quot;: {
                    &quot;type&quot;: &quot;string&quot;
                  }
                }
              }</schema>
            </output>
          </step>
        </steps>
      </workflow>
    </workflows>
  </skill>
</skills>
"
`;

exports[`assistant starts workflow 2`] = `
{
  "backstories": [
    "Backstory 1",
    "Backstory 2",
    "Backstory 3",
    "Backstory 4",
    "Backstory 5",
    "Backstory 6",
    "Backstory 7",
    "Backstory 8",
    "Backstory 9",
    "Backstory 10",
  ],
  "character": "John Doe",
  "enhanced_backstories": [
    {
      "story": "Enhanced backstory 1",
      "title": "Backstory 1",
    },
    {
      "story": "Enhanced backstory 2",
      "title": "Backstory 2",
    },
    {
      "story": "Enhanced backstory 3",
      "title": "Backstory 3",
    },
    {
      "story": "Enhanced backstory 4",
      "title": "Backstory 4",
    },
    {
      "story": "Enhanced backstory 5",
      "title": "Backstory 5",
    },
    {
      "story": "Enhanced backstory 6",
      "title": "Backstory 6",
    },
    {
      "story": "Enhanced backstory 7",
      "title": "Backstory 7",
    },
    {
      "story": "Enhanced backstory 8",
      "title": "Backstory 8",
    },
    {
      "story": "Enhanced backstory 9",
      "title": "Backstory 9",
    },
    {
      "story": "Enhanced backstory 10",
      "title": "Backstory 10",
    },
  ],
  "name": "John",
  "surname": "Doe",
  "surnames": [
    "Doe",
    "Smith",
    "Lan",
  ],
  "surnamesLength": 2,
}
`;

exports[`assistant starts workflow 3`] = `
[
  "Prompt:
Produce 2 surnames that sound good with the name: John
Context:
[
  "surnamesLength",
  "name"
]
Response:
{
  "surnames": [
    "Doe",
    "Smith",
    "Lan"
  ]
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Given the name: John select one of the surnames: Doe,Smith,Lan
Context:
[
  "name",
  "surnames"
]
Response:
{
  "surname": "Doe"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Generate a character with the name: John and the surname: Doe
Context:
[
  "name",
  "surname"
]
Response:
{
  "character": "John Doe"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Generate 10 backstories for the character: John Doe
Context:
[
  "character"
]
Response:
{
  "backstories": [
    "Backstory 1",
    "Backstory 2",
    "Backstory 3",
    "Backstory 4",
    "Backstory 5",
    "Backstory 6",
    "Backstory 7",
    "Backstory 8",
    "Backstory 9",
    "Backstory 10"
  ]
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 1
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 1",
  "story": "Enhanced backstory 1"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 2
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 2",
  "story": "Enhanced backstory 2"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 3
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 3",
  "story": "Enhanced backstory 3"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 4
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 4",
  "story": "Enhanced backstory 4"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 5
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 5",
  "story": "Enhanced backstory 5"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 6
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 6",
  "story": "Enhanced backstory 6"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 7
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 7",
  "story": "Enhanced backstory 7"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 8
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 8",
  "story": "Enhanced backstory 8"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 9
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 9",
  "story": "Enhanced backstory 9"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
  "Prompt:
Enhance the backstory: Backstory 10
Context:
[
  "backstory"
]
Response:
{
  "title": "Backstory 10",
  "story": "Enhanced backstory 10"
}
Assistant Config:
{
  "name": "Movie builder",
  "skills": [
    "movie-building-skill",
    "world-building-skill"
  ]
}
Model:
{
  "name": "gpt-4o",
  "temperature": 0.5,
  "max_tokens": 1000
}
------------------------------------------


",
]
`;
